src/Ship.js

```diff
@@ -0,0 +1,41 @@
+class Ship {
+  constructor(itinerary) {
+    this.itinerary = itinerary
+    this.currentPort = itinerary.ports[0]
+    this.currentPort.addShip(this)
+    this.previousPort = null
+  }
+  setSail() {
+    if (!this.currentPort) {
+      throw new Error("The ship is not docked")
+    }
+
+    const hasNextPort =
+      this.itinerary.ports.indexOf(this.currentPort) + 1 <
+      this.itinerary.ports.length
```

Comment:
This is a little hard to read in this format, it might look a bit long but its fine to put all this on one line
```
      this.itinerary.ports.indexOf(this.currentPort) + 1 < this.itinerary.ports.length
```